envoy/service/auth/v2/external_auth.proto

```diff
@@ -26,4 +26,22 @@ message CheckRequest {
 message CheckResponse {
   // Status `OK` allows the request. Any other status indicates the request should be denied.
   google.rpc.Status status = 1;
+
+  // An optional message which contains the attributes of an HTTP response object. This message is
```

Comment:
This PR doesn't make clear how exactly this will work. Does the HTTP service send back a proto with this? Or is this an enhancement for the normal gRPC authz service to provide it the same capabilities as the HTTP authz service?

Reply:
@htuch This should be an enhancement for the normal gRPC authz service. For an HTTP service, we just need the response. I guess that is where all my confusion comes. I think that CheckResponse just needs two extra fields: HTTP status and body. Thoughts?

Reply:
I think it's definitely the right thing for the gRPC authz service to make it explicit here. Metadata is really about the channel between the gRPC authz server and Envoy, it shouldn't be used to provide additional response information.